### ISSUE TITLE
Naked edges with only rectangles and extrusion

### ISSUE THREAD

**Issue by kristianlm** [solvespace/solvespace#1452]
Hi, and thanks for this great "little" piece of software with so much punch. SolveSpace is the first and only CAD I've learned to use.

I've got a situation that requires me to force everything to triangle meshes which doesn't seem right. Every object is 90 degrees as I'm only using rectangles. I've managed to reproduce this on the Arch Linux AUR package and on my GNU Guix system. I don't think it's a packaging issue.

### System information

- **SolveSpace version**: 3.1~70bde63c
- **Operating system:** Reproduced on Arch Linux and GNU Guix

### Expected behavior

![image](https://github.com/solvespace/solvespace/assets/259445/a2213a8a-1943-4ac4-aa81-b343da9a5a29)

I've forced `NURBS to triangle mesh` is on to get that screenshot

### Actual behavior

![image](https://github.com/solvespace/solvespace/assets/259445/71459e71-9096-4517-8830-a13981da4c18)

When `Force NURBS to triangle mesh` off, I'm missing an entire side surface.

### Additional information

[bug-report-naked-edges-with-only-cuboids-extrude-and-unions.zip](https://github.com/solvespace/solvespace/files/14521863/bug-report-naked-edges-with-only-cuboids-extrude-and-unions.zip)

The model is easily produced as follows:

![image](https://github.com/solvespace/solvespace/assets/259445/36c2d650-4da7-4ec2-9a83-d14f3cf8b7c6)

- Make rectangle A
- Extrude it downwards
- Make rectangle B in new sketch using previous sketch workplane or P
- Extrude it upwards
- Make rectangle C in new sketch using B's equivalent of point P (selecting its two connected lines)
- Extrude it upwards
- Make rectangle D in new sketch using P and its two perpendicular lines (or point on the other two cuboids)
- Extrude it right

The extrusion on A is in the opposite direction from B's extrusion. I cannot reproduce this bug if both are extruded in the same direction. The bug depends on the height of D: If D touches just A and B everything is ok. What's very strange is that if D if just _a little bit_ above C, it still works:
![image](https://github.com/solvespace/solvespace/assets/259445/d050cf02-b89f-4db5-90b4-25f98d1e8bbe)
Here, D touches C just a tiny bit and `force NURBS to triangle mesh` is off!

**Comment by phkahler** [comment]
@kristianlm can you provide a simple .slvs file demonstrating this?

**Comment by kristianlm** [comment]
> @kristianlm can you provide a simple .slvs file demonstrating this?

Indeed, it is there in my original post, though perhaps not as clearly marked as it could have been.

**Comment by ruevs** [comment]
Thank you for the nice example. It falls in the second group here https://github.com/solvespace/solvespace/issues/738, I've added it.

**Comment by ruevs** [comment]
Interestingly https://github.com/solvespace/solvespace/issues/1452 does **not** fix this.
